### PR TITLE
fix: trim slot when passed a string

### LINKED ISSUE #202: Passing raw HTML to a slot mounting option breaks when starting with a carriage return
Odd error with this:

```
const ComponentWithSlots = {
  template: `
    <div class="scoped">
      <slot name="scoped" v-bind="{ msg }" />
    </div>
  `,
  data() {
    return {
      msg: 'world'
    }
  }
}

test('scoped slots', () => {
  const wrapper = mount(ComponentWithSlots, {
    slots: {
      scoped: `
        <template #scoped="params">
        Hello {{ params.msg }}
        </template>
      `
    }
  })

  expect(wrapper.html()).toContain('Hello world')
})
```

Specifically:

BAD:

```
 scoped: `
        <template #scoped="params">
        Hello {{ params.msg }}
        </template>
      `
```

GOOD:

```
 scoped: `<template #scoped="params">
        Hello {{ params.msg }}
        </template>
      `
```

Probably just need to do `trim`.

### PR DESCRIPTION
**Pull request by lmiller1990**
resolves #202

### CODE REVIEW

File: src/mount.ts [hunk 1]
```diff
@@ -236,7 +236,7 @@ export function mount(
 
         if (typeof slot === 'string') {
           // slot is most probably a scoped slot string or a plain string
-          acc[name] = (props: VNodeProps) => h(processSlot(slot), props)
+          acc[name] = (props: VNodeProps) => h(processSlot(slot.trim()), props)
```
Comment (cexbrayat) [inline src/mount.ts:239]: nit: maybe `trim` should be called in `processSlot` (as if someone uses it in an other place of VTU, we don't have the same issue another time)

**Review by cexbrayat — APPROVED** [review]
I'm curious why this is needed. Does it mean that `compile` from core has an issue?

Otherwise, LGTM

File: src/mount.ts (hunk 1 above)
Comment (lmiller1990) [reply src/mount.ts:239]: @cexbrayat I am curious too... I had a little look but it wasn't obvious why this was causing and issue. I guess no one has tried calling `compile` where the first character is a carriage return 🤔 

I wonder if this should be fixed in core.

Comment (lmiller1990) [reply src/mount.ts:239]: Ohh I bet this is the issue https://github.com/vuejs/vue-test-utils-next/blob/e92bdcb430b6098204d20c5ea3b9c80bd21ddba7/src/utils/compileSlots.ts#L5

`startsWith('<template')` is not the same as `startsWith('\ntemplate')`.

Comment (lmiller1990) [reply src/mount.ts]: I made your suggested change! thanks.

**Pull request merged**